packages/global-styles/src/components/global-styles-provider/index.tsx

```diff
@@ -17,7 +17,11 @@ const cleanEmptyObject = ( object: any ) => {
 	return isEmpty( cleanedNestedObjects ) ? undefined : cleanedNestedObjects;
 };
 
-const useGlobalStylesUserConfig = () => {
+const useGlobalStylesUserConfig = (): [
+	boolean,
+	GlobalStylesObject,
+	( config: object ) => unknown
+] => {
```

Comment:
Typescript doesn't infer tuples very well unless you give it some help, so this is mostly a drive-by improvement for types in the file.